sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/ChangeFeedProcessorState.java

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.models;
+
+import com.azure.cosmos.ChangeFeedProcessor;
+import com.azure.cosmos.util.Beta;
+
+import java.time.Instant;
+
+/**
+ * Specifies the {@link ChangeFeedProcessor} state for a particular lease/worker.
+ */
+@Beta(Beta.SinceVersion.V4_2_0)
+public class ChangeFeedProcessorState {
+    private String id;
+    private String leaseToken;
+    private String hostName;
+    private Instant lastUpdate;
+    private String continuationToken;
+    private Instant continuationTokenTimestamp;
+    private int estimatedLag;
+
+    /**
+     * Gets the ID of the lease item representing the persistent state of a change feed processor worker.
+     *
+     * @return the ID of the lease item.
+     */
+    public String getId() {
```

Comment:
This would need to be of type FeedRange soon. https://github.com/FabianMeiswinkel/azure-sdk-for-java/blob/master/sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/FeedRange.java

I think we have 3 Options:
1) Keep it String here, to avoid breaking change use the FeedRange.toSjon as the value and expect customers to do the transformations upstream
2) Make a breaking change to this preview contract when I merge the FeedRange changes - would probably be good to set expectations with Walmart that we expect some breaking changes here
3) Add FeedRange as an additional field - make Id the FeedRange.toJson() - this way it doesn't break and the Id might not be very useful anymore but also doesn't really cause harm

I think 3) is an acceptable solution - le me know whether you agree/disagree

Reply:
The ID here is just a regular unique identifier for the lease document inside the lease collection. In the CFP "push" case we can have multiple independent instances operating on the same feed container and possibly sharing the same lease container to check-point the state; to differentiate between then a "lease prefix' is being used.

Regarding the FeedRage, I thin that will more or less matches the lease token which defines the actual scope of a single worker which will process the feeds.